classes/form/setup_form.php

```diff
@@ -75,11 +75,21 @@ protected function definition() {
         $this->_form->addElement('select', 'secondweight', get_string('field_secondweight', 'tool_blocksmanager'), $weightoptions);
         $this->_form->hideIf('secondregion', 'reposition', 'eq', 0);
         $this->_form->hideIf('secondweight', 'reposition', 'eq', 0);
+
+        $this->_form->addElement('selectyesno', 'showinsubcontexts', get_string('field_showinsubcontexts', 'tool_blocksmanager'));
+        $this->_form->addHelpButton('showinsubcontexts', 'field_showinsubcontexts', 'tool_blocksmanager');
+        $this->_form->addElement('text', 'pagetypepattern', get_string('field_pagetypepattern', 'tool_blocksmanager'));
+        $this->_form->addHelpButton('pagetypepattern', 'field_pagetypepattern', 'tool_blocksmanager');
+        $this->_form->setType('pagetypepattern', PARAM_TEXT);
+        $this->_form->setDefault('pagetypepattern', 'course-view-*');
+
         $this->_form->addElement('button', 'addline', get_string('addnew', 'tool_blocksmanager'));
+
```

Comment:
Should probably remove the extra new lines to keep the commit cleaner. Also line 92